File: 1189. Maximum Number of Balloons/1189. Maximum Number of Balloons.py
class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        letters = {"b": 0, "a": 0, "l": 0, "o": 0, "n": 0}

        for char in text:
            if char in letters:
                letters[char] += 1

        return min(letters["b"], letters["a"], letters["l"] // 2, letters["o"] // 2, letters["n"])

File: 1189. Maximum Number of Balloons/1189. Maximum Number of Balloons.java
import java.util.HashMap;
import java.util.Map;

class Solution {

  public int maxNumberOfBalloons(String text) {
    Map<Character, Integer> letters = new HashMap<>();
    letters.put('b', 0);
    letters.put('a', 0);
    letters.put('l', 0);
    letters.put('o', 0);
    letters.put('n', 0);

    for (char c : text.toCharArray()) {
      if (letters.containsKey(c)) {
        letters.put(c, letters.get(c) + 1);
      }
    }

    return Math.min(
      letters.get('b'),
      Math.min(
        letters.get('a'),
        Math.min(
          letters.get('l') / 2,
          Math.min(letters.get('o') / 2, letters.get('n'))
        )
      )
    );
  }
}
